source/dotnet/Test/AdaptiveCards.Test/AdaptiveCardApiTests.cs

```diff
@@ -108,5 +108,116 @@ public void TestParsingInvalidCardMissingType()
 }";
             Assert.ThrowsException<AdaptiveSerializationException>(() => AdaptiveCard.FromJson(json));
         }
+
+        [TestMethod]
+        // Make sure resource information for all Images and Medias in a card are returned
+        // TODO: Add test for Media information when Media type is added
+        public void TestResourceInformation()
+        {
+            var json = @"{
+  ""type"": ""AdaptiveCard"",
+  ""version"": ""1.0"",
+  ""fallbackText"": ""Fallback Text"",
+  ""speak"": ""Speak"",
+  ""backgroundImage"": ""http://adaptivecards.io/content/cats/1.png"",
+  ""body"": [
+    {
+      ""type"": ""TextBlock"",
+      ""text"": ""Hello""
+    },
+    {
+      ""type"": ""Image"",
+      ""url"": ""http://adaptivecards.io/content/cats/2.png""
+    },
+    {
+      ""type"": ""Container"",
+      ""items"": [
+        {
+          ""type"": ""Image"",
+          ""url"": ""http://adaptivecards.io/content/cats/3.png""
+        },
+        {
+          ""type"": ""ColumnSet"",
+          ""columns"": [
+            {
+              ""type"": ""Column"",
+              ""items"": [
+                {
+                    ""type"": ""Image"",
+                    ""url"": ""http://adaptivecards.io/content/cats/2.png""
+                }
+              ]
+            }
+          ]
+        },
+        {
+          ""type"": ""ImageSet"",
+          ""images"": [
+            {
+              ""type"": ""Image"",
+              ""url"": ""content/cats/4.png""
+            }
+          ]
+        }
+      ]
+    }
+  ],
+  ""actions"": [
+    {
+      ""type"": ""Action.ShowCard"",
+      ""card"": {
+          ""type"": ""AdaptiveCard"",
+          ""version"": ""1.0"",
+          ""body"": [
+            {
+              ""type"": ""TextBlock"",
+              ""text"": ""Hello""
+            },
+            {
+              ""type"": ""Image"",
+              ""url"": ""http://adaptivecards.io/content/cats/5.png""
+            }
+          ]
+      }
+    }
+  ]
+}";
+            var card = AdaptiveCard.FromJson(json).Card;
+            var expected = new RemoteResourceInformation[]
+            {
+                new RemoteResourceInformation(
+                    "http://adaptivecards.io/content/cats/1.png",
+                    typeof(AdaptiveImage),
+                    null
+                ),
+                new RemoteResourceInformation(
+                    "http://adaptivecards.io/content/cats/2.png",
+                    typeof(AdaptiveImage),
+                    null
+                ),
+                new RemoteResourceInformation(
+                    "http://adaptivecards.io/content/cats/3.png",
+                    typeof(AdaptiveImage),
+                    null
+                ),
+                new RemoteResourceInformation(
+                    "http://adaptivecards.io/content/cats/2.png",
```

Comment:
what is expected behavior when we have multiple of same resources? although we are not getting the resources, it seems wasteful to have redundant resources listed.

Reply:
Yeah I think returning everything in order (current solution) is the safer way, since the caller can process the array after getting the result in any way they want. Do you know what other renderers are handling it?